stdlib/src/collections/dict.mojo

```diff
@@ -644,6 +644,44 @@ struct Dict[K: KeyElement, V: CollectionElement](
             return default.value()[]
         raise "KeyError"
 
+    fn popitem(inout self) raises -> DictEntry[K, V]:
+        """Remove and return a (key, value) pair from the dictionary. Pairs are returned in LIFO order.
+
+        popitem() is useful to destructively iterate over a dictionary, as often used in set algorithms.
+        If the dictionary is empty, calling popitem() raises a KeyError.
+
+        Args: None
+
+        Returns:
+            Last dictionary item
+
+        Raises:
+            "KeyError" if the dictionary is empty.
+        """
+        # Get the last key so we can get the slot and set it to REMOVED
+        for key in Self.__reversed__(self):
```

Comment:
I think the code here is not quite correct, it will take the first value, but what we need is the last item.

But your overall idea I think it quite nice, I should get the last item and just call `pop` instead. 👍

Reply:
> it will take the first value, but what we need is the last item.

I'm not sure I understand this comment, in the code snippet, isn't using `__reverse__` making us start with the last element?

Reply:
```mojo
var key_found: Optional[K] = None
var value_found: Optional[V] = none
for key_value in self.items().__reverse__():
    key_found = key_value.key[]
    value_found = key_value.value[]
```

yes, it starts from the last element, so `key_found` and `value_found` is replaced with the first one.